src/controllers/confirmation.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { NextFunction, Request, Response } from "express";
+import {formatDateForDisplay} from "../client/date.formatter";
+import logger from "../logger";
+import {PTFCompanyProfile} from "../model/company.profile";
+import * as templatePaths from "../model/template.paths";
+import * as sessionService from "../services/session.service";
+import {ALREADY_SUBMITTED, COMPANY_NUMBER, COMPANY_PROFILE, SIGN_IN_INFO, USER_PROFILE} from "../session/keys";
+import {IUserProfile} from "../session/types";
+
+const ACCOUNTS_NEW_DEADLINE: number = 28;
+const CS_NEW_DEADLINE: number = 14;
+
+const createMissingError = (item: string): Error => {
+    const errMsg: string = item + " missing from session";
+    return new Error(errMsg);
+};
+
+const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const companyProfile: PTFCompanyProfile =
+      await sessionService.getPromiseToFileSessionValue(req.chSession, COMPANY_PROFILE);
+  const signInInfo = req.chSession.getSignedInInfo();
+  const userProfile: IUserProfile = signInInfo[USER_PROFILE] as IUserProfile;
+
+  const email = userProfile.email;
+  const dateRequested: Date = new Date(Date.now());
+  const extensionPeriodInDays =
+      (companyProfile.isAccountsOverdue) ? ACCOUNTS_NEW_DEADLINE : CS_NEW_DEADLINE;
+  const extension: Date = new Date(dateRequested);
+  extension.setDate(dateRequested.getDate() + extensionPeriodInDays);
+
+  const isSubmitted: boolean = await sessionService.getPromiseToFileSessionValue(req.chSession, ALREADY_SUBMITTED);
+  const token: string = req.chSession.accessToken() as string;
+
+  if (!(isSubmitted && token)) {
```

Comment:
i think the if check should just be on isSubmitted. We don't need token until we call the api so the get token can go inside the if and it will only be called when we need it. Also, if submitted = false and token is false then it will attempt to call the api but fail as token is false.

Reply:
Addressed for comment 2 above

Reply:
Made the commit this complicated: Removed both submitted and token checks for now as they are not needed immediately will not add anything that cannot be tested first so as to avoid any mistakes.